### PR TITLE
Floating UI components

### PR DESCRIPTION
**Pull request by mbarrenechea**
This PR adds a new set of common floating UI components

- [x] Tooltip
- [x] Modal

### CODE REVIEW

**Comment by andresgnlez** [comment]
image

In the modal with a fixed header, when you scroll down, there's a pixel line displaying the faded content.

File: src/components/modal/component.tsx [hunk 1]
```diff
@@ -1,31 +1,95 @@
...
+            >
+              <FloatingFocusManager context={context}>
+                <>
+                  <Media
+                    lessThan="sm"
+                    className="absolute flex flex-col w-full h-full pointer-events-none grow"
+                  >
+                    <ModalContent
+                      {...props}
+                      viewport=""
```
Comment (andresgnlez) [inline src/components/modal/component.tsx:70]: I think a better approach would be to change the `viewport` type to optional and remove this empty viewport. Or typing `viewport` as `viewport?: 'sm' | undefined` makes more sense, doesn't it?

**Review by andresgnlez — CHANGES_REQUESTED** [review]

File: src/components/modal/component.tsx (hunk 1 above)
Comment (mbarrenechea) [reply src/components/modal/component.tsx]: Done!

**Comment by mbarrenechea** [comment]
> image
> 
> In the modal with a fixed header, when you scroll down, there's a pixel line displaying the faded content. This also happens at the bottom of the content.

It's something related to the browser (Chrome and Safari, Firefox works well), it happened to me in the past. 
I fixed it by adding en extra div with 1px padding on the y axis and move the gradients there... 🤷‍♂️

**Pull request merged**